apps/snitch_core/test/data/schema/payment/card_payment_test.exs

```diff
@@ -13,18 +26,100 @@ defmodule Snitch.Data.Schema.CardPaymentTest do
 
   describe "CardPayment records" do
     test "refer card type Payments (uniquely)", context do
-      %{ccd: card} = context
-      card_payment = CardPayment.changeset(%CardPayment{}, %{payment_id: card.id}, :create)
+      %{ccd: card_payment, user: user} = context
+
+      card_payment =
+        CardPayment.changeset(
+          %CardPayment{},
+          %{payment_id: card_payment.id, card: Map.put(@card, :user_id, user.id)},
+          :create
+        )
+
       assert {:error, %{errors: errors}} = Repo.insert(card_payment)
       assert errors == [payment_id: {"has already been taken", []}]
     end
 
     test "DON'T refer other Payments", context do
-      %{chk: check} = context
-      card_payment = CardPayment.changeset(%CardPayment{}, %{payment_id: check.id}, :create)
-      Repo.insert(card_payment)
+      %{chk: check, user: user} = context
+
+      card_payment =
+        CardPayment.changeset(
+          %CardPayment{},
+          %{payment_id: check.id, card: Map.put(@card, :user_id, user.id)},
+          :create
+        )
+
       assert {:error, %Ecto.Changeset{errors: errors}} = Repo.insert(card_payment)
       assert errors == [payment_id: {"does not refer a card payment", []}]
     end
+
+    test "with all good", context do
+      %{user: user, card_method: card_m, order: order} = context
+
+      payment_struct = %Payment{@payment | payment_method_id: card_m.id, order_id: order.id}
+
+      payment = insert(payment_struct)
+
+      card_payment =
+        CardPayment.changeset(
+          %CardPayment{},
+          %{payment_id: payment.id, card: Map.put(@card, :user_id, user.id)},
+          :create
+        )
+
+      %{valid?: validity} = card_payment
+
+      assert validity
+      assert {:ok, _} = Repo.insert(card_payment)
+    end
+
+    test "with already having card_id", context do
```

Comment:
I don't understand the point of this test. Did you want to test if the same card can be used for 2 orders? I don't think that needs to be tested as the database design guarantees that.
The use of `:rand` in this test is wrong, instead you should use the factory methods to insert dummy `orders`, `payments` and `card_payments`.

Reply:
I want to test the `assoc_card` function for `card_payment` schema. Because when we are having card_id (card is previously used & saved) then it wont make association again. So if card_id is present the `else` part of `assoc_card` will be executed. To to meet the coverage for the else part I have wrote this test.

Reply:
I see that you want to cover the `else` block, but why do you have to _"use a card in a previous payment"_ to test this? All this test requires is that you:
1. create a `card` using its factory method,
2. create a `card_payment` as done on [L109](https://github.com/aviabird/snitch/pull/45/files#diff-f65d50d77a98282f5baf6a439233a14aR109)